src/nvim/main.c

```diff
@@ -259,6 +261,35 @@ int main(int argc, char **argv)
   command_line_scan(&params);
   server_init(params.listen_addr);
 
+  if (params.cs_remote) {
+    const char *env_addr = os_getenv("NVIM_LISTEN_ADDRESS");
```

Comment:
For now I think we should require an explicit `--server {addr}` command line option when any of the `--remote` stuff is used. (Vim has `--servername` but we don't need to exactly match the name because we aren't exactly matching Vim's _behavior_ anyways.)

Later we can discuss the "implicit" server connections, that's part of the design discussion related to "discovery". But for now, adding this extra meaning of `$NVIM_LISTEN_ADDRESS` is too much to think about it.

Reply:
Using `nvr --remote-cmd ...` inside a nvim `:terminal` is very useful. Even if this doesn't match that behaviour directly, I certainly would expect something more convenient than spelling out

    nvim --server $NVIM_LISTEN_ADDRESS --remote-cmd ...

every time. (like say an option for "parent nvim")

Reply:
@bfredl That will be the purpose of $NVIM which I will implement soon. But adding more purpose to `$NVIM_LISTEN_ADDRESS` is something I would like to avoid.

Reply:
The purpose is already there but that doesn't mean we need to plant more roots by adding support in `--remote`. There is no back-compt or legacy support issue because `--remote` is new. Anyone using `--remote` will already have `$NVIM`.

Reply:
> Anyone using --remote will already have $NVIM.

This is true enough if we commit to add `--remote` and `$NVIM`  in the same release. Do we do that?

Reply:
Yes, definitely.

Reply:
One more thing about 
```c
const char *env_addr = os_getenv("NVIM_LISTEN_ADDRESS");
```
if you don't provide `NVIM_LISTEN_ADDRESS` it somehow takes the address of itself, I'll inspect how it happens(not related).

About the changes, 
I'll add `--server {addr}` command line option and remove getting server address from `NVIM_LISTEN_ADDRESS` as discussed above.

Reply:
For reference, $NVIM added in https://github.com/neovim/neovim/pull/11009